config/src/main/java/org/springframework/security/config/annotation/authentication/configuration/InitializeAuthenticationProviderBeanManagerConfigurer.java

```diff
@@ -49,16 +56,33 @@ public void init(AuthenticationManagerBuilder auth) throws Exception {
 
 	class InitializeAuthenticationProviderManagerConfigurer extends GlobalAuthenticationConfigurerAdapter {
 
+		private final Log logger = LogFactory.getLog(getClass());
+
 		@Override
 		public void configure(AuthenticationManagerBuilder auth) {
 			if (auth.isConfigured()) {
 				return;
 			}
-			AuthenticationProvider authenticationProvider = getBeanOrNull(AuthenticationProvider.class);
-			if (authenticationProvider == null) {
+			List<BeanWithName<AuthenticationProvider>> authenticationProviders = getBeansWithName(
+					AuthenticationProvider.class);
+			if (authenticationProviders.isEmpty()) {
 				return;
 			}
+			else if (authenticationProviders.size() > 1) {
+				List<String> beanNames = authenticationProviders.stream().map(BeanWithName::getName).toList();
+				this.logger.warn(LogMessage.format(
+						"Found %s AuthenticationProvider beans, with names %s. "
```

Comment:
While informative, would the developer know what to do with this message? I wonder if it should also say something like "Consider publishing only one authentication provider bean or wiring them directly to the DSL".

Reply:
I went with:

> Found 3 AuthenticationProvider beans, with names [x, y, z].  Global Authentication Manager will not be configured with AuthenticationProviders.  Consider publishing a single AuthenticationProvider bean, or wiring your Providers directly using the DSL.